tests/observability/micrometer/src/test/java/org/wildfly/test/cloud/observability/micrometer/MicrometerIT.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ *  Copyright 2022 Red Hat, Inc., and individual contributors
+ *  as indicated by the @author tags.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.wildfly.test.cloud.observability.micrometer;
+
+import static org.wildfly.test.cloud.common.WildflyTags.KUBERNETES;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import jakarta.ws.rs.core.MediaType;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.LocalPortForward;
+import io.restassured.RestAssured;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.wildfly.test.cloud.common.KubernetesResource;
+import org.wildfly.test.cloud.common.WildFlyCloudTestCase;
+import org.wildfly.test.cloud.common.WildFlyKubernetesIntegrationTest;
+
+/**
+ * @author <a href="mailto:jasondlee@redhat.com">Jason Lee</a>
+ */
+@Tag(KUBERNETES)
+@WildFlyKubernetesIntegrationTest(
+        namespace = "micrometer",
+        kubernetesResources = {
+                @KubernetesResource(definitionLocation = "src/test/container/collector-config.yml"),
+                @KubernetesResource(definitionLocation = "src/test/container/opentelemetry-collector.yml"),
+                @KubernetesResource(definitionLocation = "src/test/container/service.yml")
+        })
+public class MicrometerIT extends WildFlyCloudTestCase {
+
+    @Test
+    public void test() throws Exception {
+        int requestCount = (int) ((Math.random() * 10) + 10);
+        getHelper().doWithWebPortForward("", url -> makeRestRequests(url, requestCount));
+
+
+        try (LocalPortForward p = getHelper().getK8sClient().services().withName("opentelemetrycollector").portForward(1234)) {
+            URI uri = new URI("http://localhost:" + p.getLocalPort() + "/metrics");
+
+            final HttpClient client = HttpClient.newBuilder().build();
+            final HttpRequest request = HttpRequest.newBuilder().uri(uri).GET().build();
+
+            boolean found = false;
+            int count = 0;
+            while (count < 10) {
```

Comment:
It's largely an arbitrary number so that it doesn't hang forever. We could bump it up or parameterize it (via system properties?), I guess, but unless the test proves to be flaky, I don't see the need just yet.

Reply:
+1, I just asked for details. Thanks!

Reply:
It _is_ a magic number, so it should make reviewers question it. ;)